src/backend/distributed/operations/shard_transfer.c

```diff
@@ -170,37 +154,23 @@ citus_copy_shard_placement(PG_FUNCTION_ARGS)
 	int32 sourceNodePort = PG_GETARG_INT32(2);
 	text *targetNodeNameText = PG_GETARG_TEXT_P(3);
 	int32 targetNodePort = PG_GETARG_INT32(4);
-	bool doRepair = PG_GETARG_BOOL(5);
-	Oid shardReplicationModeOid = PG_GETARG_OID(6);
+	Oid shardReplicationModeOid = PG_GETARG_OID(5);
 
 	char *sourceNodeName = text_to_cstring(sourceNodeNameText);
 	char *targetNodeName = text_to_cstring(targetNodeNameText);
 
 	char shardReplicationMode = LookupShardTransferMode(shardReplicationModeOid);
-	if (doRepair && shardReplicationMode == TRANSFER_MODE_FORCE_LOGICAL)
-	{
-		ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
-						errmsg("logical replication cannot be used for repairs")));
-	}
 
 	ShardInterval *shardInterval = LoadShardInterval(shardId);
 	ErrorIfTableCannotBeReplicated(shardInterval->relationId);
 	ErrorIfTargetNodeIsNotSafeToCopyTo(targetNodeName, targetNodePort);
+	EnsureNoModificationsHaveBeenDone();
```

Comment:
Why is this needed now?

Reply:
this check was missing from regular shard copy, which was a bug revealed by rewriting one of the tests from doing repair to regular copy